### PR TITLE
Use providerID as a link between Node and a Machine

### PR DESCRIPTION
**Pull request by lin007pro**

### CODE REVIEW

File: cmd/kubeturbo/app/kubeturbo_builder.go [hunk 1]
```diff
@@ -254,7 +258,8 @@ func (s *VMTServer) Run() {
 		WithDiscoveryInterval(s.DiscoveryIntervalSec).
 		WithValidationTimeout(s.ValidationTimeout).
 		WithValidationWorkers(s.ValidationWorkers).
-		WithSccSupport(s.sccSupport)
+		WithSccSupport(s.sccSupport).
+		WithCAPINapespace(s.ClusterAPINamespace)
```
Comment (ading1977) [inline cmd/kubeturbo/app/kubeturbo_builder.go]: Please correct typo

File: pkg/kubeturbo_service_config.go [hunk 2]
```diff
@@ -99,3 +100,8 @@ func (c *Config) WithSccSupport(sccSupport []string) *Config {
 	c.SccSupport = sccSupport
 	return c
 }
+
+func (c *Config) WithCAPINapespace(CAPINamespace string) *Config {
```
Comment (ading1977) [inline pkg/kubeturbo_service_config.go]: Please correct typo

File: pkg/action/action_handler.go [hunk 3]
```diff
@@ -46,9 +46,10 @@ type ActionHandlerConfig struct {
 	kubeletClient  *kubeclient.KubeletClient
 	StopEverything chan struct{}
 	sccAllowedSet  map[string]struct{}
+	cAPINamespace  string
 }
 
-func NewActionHandlerConfig(kubeClient *client.Clientset, kubeletClient *kubeclient.KubeletClient, sccSupport []string) *ActionHandlerConfig {
+func NewActionHandlerConfig(cApiNamespace string, kubeClient *client.Clientset, kubeletClient *kubeclient.KubeletClient, sccSupport []string) *ActionHandlerConfig {
```
Comment (ading1977) [inline pkg/action/action_handler.go]: Maybe move the **cApiNamespace** to the last of the function parameter?

File: cmd/kubeturbo/app/kubeturbo_builder.go (hunk 1 above)
Comment (lin007pro) [reply cmd/kubeturbo/app/kubeturbo_builder.go]: Hi,
I don't see a typo. I agree we need to allow for the logic to handle other providers, although the Cluster API is not provider specific from coding standpoint.
I propose to address this when we know more.

File: pkg/action/action_handler.go (hunk 3 above)
Comment (lin007pro) [reply pkg/action/action_handler.go]: I prefer to have it as a first one. The emphasis is important, IMHO.

**Pull request merged**